### DISCUSSION TITLE
updating edge embeddings

### DISCUSSION THREAD

**Discussion by errhernandez** [pyg-team/pytorch_geometric#4335] [Q&A]
Hello. I have noticed that base class MessagePassing has functions edge_updater() and edge_update() which, if my understanding is correct seem to play the same roles as propagate() and update() do for nodes, but for edges instead. Is this so? Does it mean that one can use these functions to update edge embeddings? I actually looked through the source to try to find examples of convolutional layers that used this facility but could find none. Yet it is a very interesting possibility, as done for example in CensNet (https://arxiv.org/abs/2010.13242).  Thanks.

**Comment by rusty1s** [comment]
Yes, that is correct. One can leverage `edge_updater()` to update edge embeddings or compute attention coefficients. We sadly do not have an example of its use yet, except for its [tests](https://github.com/pyg-team/pytorch_geometric/blob/master/test/nn/conv/test_message_passing.py#L164-L180) (`GATConv` might be a good one). Happy to help out if you want to apply it or want to contribute an example of it :)

**Reply by errhernandez** [reply]
Thank you for your response. I will try my hand at implementing CensNet with this; I will let you know how it goes.

**Reply by rusty1s** [reply]
Just modified `GATConv` to make use of `edge_updater`: https://github.com/pyg-team/pytorch_geometric/pull/4342
